src/object.d

```diff
@@ -3241,6 +3241,226 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+int _adCmpT(T1, T2)(T1[] s1, T2[] s2)
+{
+    // U1 gets the unqualified version of T1
+    static if (is(T1 W == immutable W)) alias U1 = W;
+    else static if (is(T1 W == const W)) alias U1 = W;
+    else alias U1 = T1;
```

Comment:
I spoke to @somzzz and advised that we should take a larger step once the bit of duplication becomes painful. Since her work will intensify the use of templates in object.d, that's likely to happen.